SPLASH/res/shaders/compute/ParticleComputeShader.hlsl

```diff
@@ -117,12 +118,12 @@ void updateAnimation(int particleIndex) {
     int v0Index = particleIndex * 6;
     
     // Hardcoded dimention of texture atlas. 
-    int textureSize = 8;                                                                    // This atlas dimension is 8x4
-    int ai = (CSPhysicsBuffer[particleIndex].animationIndexCounter += inputBuffer.frameTime*24.f) % (4 * textureSize);
+    int atlasSizeX = inputBuffer.atlasSize.x;
+    int ai = (CSPhysicsBuffer[particleIndex].animationIndexCounter += inputBuffer.frameTime*24.f) % (inputBuffer.atlasSize.y * atlasSizeX);
 
-    float offset = 1.0f / textureSize;
-    float ix = ai % textureSize;
-    float iy = (int)(ai / textureSize);
+    float offset = 1.0f / atlasSizeX;
+    float ix = ai % atlasSizeX;
+    float iy = (int)(ai / atlasSizeX);
```

Comment:
ai and atlasSizeX are both int, so this is probably nothing important, but feels a bit wonky with all the casting and division between types